lms/djangoapps/courseware/module_render.py

```diff
@@ -270,10 +276,15 @@ def get_module_for_descriptor(user, request, descriptor, field_data_cache, cours
 
     user_location = getattr(request, 'session', {}).get('country_code')
 
+    student_kvs = DjangoKeyValueStore(field_data_cache)
+    if is_masquerading_as_specific_student(user, course_key):
+        student_kvs = MasqueradingKeyValueStore(student_kvs, request.session)
+    student_data = KvsFieldData(student_kvs)
+
     return get_module_for_descriptor_internal(
         user=user,
         descriptor=descriptor,
-        field_data_cache=field_data_cache,
+        student_data=student_data,
```

Comment:
Oh, nice. I love that this limits knowledge of the `FieldDataCache` interface even more.